packages/wrangler/src/dev.tsx

```diff
@@ -597,3 +541,69 @@ function memoizeGetPort(defaultPort: number) {
 		return portValue || (portValue = await getPort({ port: defaultPort }));
 	};
 }
+
+async function getBindings(
+	configParam: Config,
+	args: AdditionalDevProps
+): Promise<CfWorkerInit["bindings"]> {
+	const bindings = {
+		kv_namespaces: [
+			...(configParam.kv_namespaces || []).map(
+				({ binding, preview_id, id: _id }) => {
+					// In `dev`, we make folks use a separate kv namespace called
+					// `preview_id` instead of `id` so that they don't
+					// break production data. So here we check that a `preview_id`
+					// has actually been configured.
+					// This whole block of code will be obsoleted in the future
+					// when we have copy-on-write for previews on edge workers.
+					if (!preview_id) {
+						// TODO: This error has to be a _lot_ better, ideally just asking
+						// to create a preview namespace for the user automatically
+						throw new Error(
+							`In development, you should use a separate kv namespace than the one you'd use in production. Please create a new kv namespace with "wrangler kv:namespace create <name> --preview" and add its id as preview_id to the kv_namespace "${binding}" in your wrangler.toml`
+						); // Ugh, I really don't like this message very much
+					}
+					return {
+						binding,
+						id: preview_id,
+					};
+				}
+			),
+			...(args.kv || []),
```

Comment:
this isn't because of your PR, but - does this mean args.kv doesn't get the same validation as above? That's ... not great. Similarly for args.durableObjects and args.vars